### ISSUE TITLE
Test failure when building a release

### ISSUE THREAD

**Issue by ArchangeGabriel** [whipper-team/whipper#420]
The logger test fails because the version number is not following the regexp pattern:
```
======================================================================
FAIL: testLogger (whipper.test.test_result_logger.LoggerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/whipper/src/whipper/whipper/test/test_result_logger.py", line 143, in testLogger
    r'Log created by: whipper '
AssertionError: Regexp didn't match: 'Log created by: whipper [\\d]+\\.[\\d]+.[\\d]+\\.dev[\\w\\.\\+]+ \\(internal logger\\)' not found in 'Log created by: whipper 0.8.0 (internal logger)'

----------------------------------------------------------------------
```
I’ve tried to `sed -i 's|\\\.dev\[\\w\\\.\\+\]+||' whipper/test/test_result_logger.py` (on the basis that it was using the actual whipper version), but then it failed with:
```
======================================================================
FAIL: testLogger (whipper.test.test_result_logger.LoggerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/whipper/src/whipper/whipper/test/test_result_logger.py", line 143, in testLogger
    r'Log created by: whipper '
AssertionError: Regexp didn't match: 'Log created by: whipper [\\d]+\\.[\\d]+.[\\d]+ \\(internal logger\\)' not found in 'Log created by: whipper 0.8.1.dev0+g4238e71.d20191030 (internal logger)'

----------------------------------------------------------------------
```
because I guess they are two tests ran with different logs. So I went for making the “dev” part optional (`sed -i 's|\\\.dev\[\\w\\\.\\+\]+|(\\.dev[\\w\\.+]+)?|' whipper/test/test_result_logger.py`).

If that changes looks reasonable to you, I can submit a PR for this.

**Comment by ArchangeGabriel** [comment]
Actually that second version number looks strange.

While checking the procuded package for the 0.8.0 release, I’ve seen this:
```
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/PKG-INFO
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/SOURCES.txt
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/dependency_links.txt
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/entry_points.txt
whipper /usr/lib/python2.7/site-packages/whipper-0.8.1.dev0+g4238e71.d20191030-py2.7.egg-info/top_level.txt
```
So there is likely an issue with the way whipper finds out the version number. Note that we had to switch from GitHub tarball to cloning the git tag since else we faced this:
```
Traceback (most recent call last):
  File "setup.py", line 28, in <module>
    'scripts/accuraterip-checksum',
  File "/usr/lib/python2.7/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 446, in __init__
    k: v for k, v in attrs.items()
  File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__
    self.finalize_options()
  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 734, in finalize_options
    ep.load()(self, ep.name, value)
  File "/build/whipper/src/whipper-0.8.0/.eggs/setuptools_scm-3.3.3-py2.7.egg/setuptools_scm/integration.py", line 17, in version_keyword
    dist.metadata.version = get_version(**value)
  File "/build/whipper/src/whipper-0.8.0/.eggs/setuptools_scm-3.3.3-py2.7.egg/setuptools_scm/__init__.py", line 150, in get_version
    parsed_version = _do_parse(config)
  File "/build/whipper/src/whipper-0.8.0/.eggs/setuptools_scm-3.3.3-py2.7.egg/setuptools_scm/__init__.py", line 113, in _do_parse
    "use git+https://github.com/user/proj.git#egg=proj" % config.absolute_root
LookupError: setuptools-scm was unable to detect version for '/build/whipper/src/whipper-0.8.0'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
```

**Comment by ArchangeGabriel** [comment]
Yeah, I can confirm there is an issue with getting the version on release builds:
```
$ whipper --version
whipper 0.8.1.dev0+g4238e71.d20191030
```

**Comment by Freso** [comment]
> If that changes looks reasonable to you, I can submit a PR for this.

Please do.

> Yeah, I can confirm there is an issue with getting the version on release builds:

Hm. This should be a separate issue. Care to open one?

**Comment by ArchangeGabriel** [comment]
@Freso [Done](https://github.com/whipper-team/whipper/pull/421) and [done](https://github.com/whipper-team/whipper/issues/422).

**Comment by ArchangeGabriel** [comment]
Now I get it while the test started failing once fixed in the first attempt: because of #422, the actual version number wasn’t 0.8.0 anymore…

**Comment by Freso** [comment]
Because the test doesn’t account for versioning of dirty trees.
